packages/substrate/src/fungible.ts

```diff
@@ -94,21 +99,41 @@ class SubstrateFungibleAssetTransfer extends BaseTransfer {
     }
   }
 
+  async verifyBalance(): Promise<void> {
+    const fee = await this.getFee();
+
+    if (!isValidAddressForNetwork(this.senderAddress, Network.SUBSTRATE))
+      throw new Error('Sender address is incorrect');
+
+    if (new BN(this.amount.toString()).lt(fee.fee)) {
+      throw new Error('Transfer amount should be higher than transfer fee');
+    }
+
+    const { data: balance } = (await this.sourceNetworkProvider.query.system.account(
+      this.senderAddress,
+    )) as unknown as {
+      data: AccountData;
+    };
+
+    // TODO: clarify if we extract fee from amount or add on top of it
+    const costs = new BN(this.amount.toString()).add(fee.fee);
```

Comment:
@saadjhk @wainola Do you know for sure how the fee will be calculated here?

Reply:
IMO `fee` is deducted from the user specified amount on substrate

c.c @MakMuftic @wainola